llvm/lib/CodeGen/MachinePipeliner.cpp

```diff
@@ -528,8 +528,16 @@ bool MachinePipeliner::useSwingModuloScheduler() {
 }
 
 bool MachinePipeliner::useWindowScheduler(bool Changed) {
-  // WindowScheduler does not work when it is off or when SwingModuloScheduler
-  // is successfully scheduled.
+  // WindowScheduler does not work for following cases:
+  // 1. when it is off.
+  // 2. when SwingModuloScheduler is successfully scheduled.
+  // 3. when pragma II is enabled.
+  if (II_setByPragma) {
+    LLVM_DEBUG(
+        dbgs() << "Window scheduling is disabled when Pragma II is set.\n");
```

Comment:
Message doesn't sound like it means anything, probably should name as the metadata?

Reply:
Is it better to change the debug message to:
```
  if (II_setByPragma) {
    LLVM_DEBUG(
        dbgs() << "Window scheduling is disabled when llvm.loop.pipeline.initiationinterval is set.\n");
  }
```

Reply:
Yes

Reply:
Updated.